src/coreComponents/mesh/generators/CellBlockManagerABC.hpp

```diff
@@ -131,6 +131,12 @@ class CellBlockManagerABC : public dataRepository::Group
    */
   virtual array2d< real64, nodes::REFERENCE_POSITION_PERM > getNodePositions() const = 0;
 
+  /**
+   * @brief Returns a mutable view of the node coordinates in a (numNodes, 3) 2d array.
+   * @return A view to the array.
+   */
+  virtual arrayView2d< real64, nodes::REFERENCE_POSITION_USD > getNodePositions() = 0;
```

Comment:
Thanks for pointing it out. I second that the setter accessors should be used more carefully in the base classes such as "CellBlockManagerABC". The main reason that we put the setters over there is because, we need to reuse the informations from CellBlockManager to modify some necessary properties for high-order mesh-levels (while utilizing some existing properties from base mesh-level for code conciseness and clarity). 

As far as I understand, in the current design, the high-order mesh-levels have their own "nodeManager", "edgeManager", and "faceManager"; however, they will share the same "CellBlockManager" with the base mesh-level. That is another reason why we need to have the setters to modify some parameters (mainly for the new nodes that are added to the high-order mesh faces, edges, and elements). 

I agree with you that "there has been some overlap in GEOS between the geometrical description of the mesh and the discretization on top of it". And we probably need to find some other ways to deal with it, but it would need some fundamental changes of the infrastructure which we are trying to avoid (to have significant modifications that might affect other solvers) in this MPI PR. 

In conclusion, I agree with your point and second that a better design of the structure is needed, while I think we could leave it to another PR with more infrastructure discussions involved.